### PR TITLE
feat: dotLottie support for android

### PR DESCRIPTION
**Pull request by matinzd**
Related to: https://github.com/lottie-react-native/lottie-react-native/pull/1071

### CODE REVIEW

File: packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewManagerImpl.kt [hunk 1]
```diff
@@ -286,4 +286,13 @@ internal object LottieAnimationViewManagerImpl {
     ) {
         viewManager.textFilters = textFilters
     }
+
+    @JvmStatic
+    fun setSourceDotLottieURI(
+        uri: String?,
+        viewManager: LottieAnimationViewPropertyManager
+    ) {
+        viewManager.sourceDotLottie = uri
```
Comment (TheRogue76) [inline packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewManagerImpl.kt:295]: Considering what react bridge does when a field goes from string to undefined on iOS, do you think we should also add the clean up here as well for Android?

Comment (TheRogue76) [reply packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewManagerImpl.kt:295]: Nit: Also please move this up under line 168 so all the source settings are in the same place

**Review by TheRogue76 — COMMENTED** [review]
Looks Good. Just two comments and the only blocker is validating that the values get reset correctly when switching from source methods

File: packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewManagerImpl.kt (hunk 1 above)
Comment (matinzd) [reply packages/core/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewManagerImpl.kt:295]: For Android we have null check everywhere `LottieAnimationViewPropertyManager`. I moving it up.

**Review by TheRogue76 — APPROVED** [review]

**Pull request merged**